app/controllers/report_controller.rb

```diff
@@ -10,35 +10,33 @@ def rsolr_request_error(exception)
   end
 
   def data
-    params[:sord] ||= "asc"
-    rows_per_page = params[:rows] ? params.delete(:rows).to_i : 10
-    params[:per_page] = rows_per_page * [params.delete(:npage).to_i, 1].max
+    params[:per_page] = params[:per_page] ? params[:per_page].to_i : Report::ROWS_PER_PAGE
+    params[:page] ||= 1
     @report = Report.new(params, current_user:)
 
     respond_to do |format|
       format.json do
         render json: {
           page: params[:page].to_i,
-          records: @report.num_found,
-          total: (@report.num_found / rows_per_page.to_f).ceil,
-          rows: @report.report_data
+          last_page: (@report.num_found / params[:per_page].to_f).ceil,
+          data: @report.report_data
         }
       end
       format.xml { render xml: @report.report_data }
     end
   end
 
-  def content_types
-  end
-
   def download
-    fields = params["fields"].present? ? params.delete("fields").split(/\s*,\s*/) : nil
-    params[:per_page] = 10
-    params[:page] = 1
-    response.headers["Content-Type"] = "application/octet-stream"
-    response.headers["Content-Disposition"] = "attachment; filename=report.csv"
+    response.headers.delete("Content-Length")
```

Comment:
Why is this necessary?

Reply:
@justinlittman This is used to let Rack know to stream the response instead of filling a buffer. Learned about this technique by doing research about how to stream large CSV results to the user, e.g., https://medium.com/reflektive-engineering/faster-csv-downloads-using-enumerator-7e9b94b870d3.